ivy/functional/frontends/paddle/nn/functional/loss.py

```diff
@@ -230,3 +230,56 @@ def margin_ranking_loss(input, other, label, margin=0.0, reduction="mean", name=
     out = ivy.atleast_1d(out)
 
     return out
+
+
+@with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def triplet_margin_loss(
+    input,
+    positive,
+    negative,
+    margin=1.0,
+    p=2.0,
+    eps=1e-06,
+    swap=False,
+    reduction="mean",
+):
+    def pairwise_distance(x1, x2, *, p=2.0, eps=1e-06, keepdim=False):
```

Comment:
This inner function better should be declared as a private function and should be better defined above this function and only be called inside this function